consensus/service/src/byzantine_ledger/worker.rs

```diff
@@ -873,47 +873,45 @@ impl<
         // The enclave cannot provide a timestamp, so this happens in untrusted.
         signature.set_signed_at(chrono::Utc::now().timestamp() as u64);
 
-        BlockData::new(block, block_contents, Some(signature))
+        let metadata = BlockMetadata::new(
+            block.id.clone(),
+            Some((&self.scp_node.quorum_set()).into()),
+            self.enclave.get_ias_report().ok(),
+        );
+        // TODO: Is this the right key for signing the metadata?
+        let metadata =
+            SignedBlockMetadata::from_contents_and_keypair(metadata, &self.msg_signer_key)
```

Comment:
Another subtle note is I don't know if from_contents_and_keypair signs %100 correctly. Comments on that in the method definition below.